### DISCUSSION TITLE
Structed output - GitHub Models

### DISCUSSION THREAD

**Discussion by rajyraman** [microsoft/semantic-kernel#9088] [Q&A] [answered]
Execution settings does not seem to be even getting used. I am trying with GitHub Models.

![image](https://github.com/user-attachments/assets/1b733fd6-aa35-4bb4-8b43-768e58bdf5ed)

Using SK 1.2.1
![image](https://github.com/user-attachments/assets/c839ce69-89a9-4747-841d-5215b989d284)

![image](https://github.com/user-attachments/assets/82619ee9-8110-4503-8f20-adc1adb0b4a9)

Model version seem to be supported.

![image](https://github.com/user-attachments/assets/5c50f10e-02de-4be3-8fe5-a03a0f339630)

**Comment by evchaki** [comment]
@dmytrostruk can you take a look at this one?

**Comment by NickDrouin** [comment]
Possibly related, my comment here: 

https://discord.com/channels/1063152441819942922/1075907964667437072/1291837574008279040

When I dug into this, it reminded me of dll-hell issues from way back when.
Looks like the API version needed to access 2024-08-01 is only available in `Azure.AI.OpenAI 2.1.0-beta.1`, but SK 1.21.1 is not linking to that.

Trying to force 2.1.0-beta.1 beaks auth (or I can't seem to get it to behave).

**Comment by dmytrostruk** [comment, marked as answer]
@rajyraman This needs further investigation, I created an issue here: https://github.com/microsoft/semantic-kernel/issues/9136
